cpu/cc2538/periph/uart.c

```diff
@@ -189,30 +129,12 @@ static int init_base(uart_t uart, uint32_t baudrate)
 {
     cc2538_uart_t *u = NULL;
 
-    switch (uart) {
-#if UART_0_EN
-        case UART_0:
-            u = UART_0_DEV;
-            gpio_init_af(UART_0_RX_PIN, UART0_RXD, GPIO_IN);
-            gpio_init_af(UART_0_TX_PIN, UART0_TXD, GPIO_OUT);
-            break;
-#endif
-#if UART_1_EN
-        case UART_1:
-            u = UART_1_DEV;
-            gpio_init_af(UART_1_RX_PIN, UART1_RXD, GPIO_IN);
-            gpio_init_af(UART_1_TX_PIN, UART1_TXD, GPIO_OUT);
-            break;
-#endif
-
-        default:
-            (void)u;
-            return UART_NODEV;
-    }
+    u = uart_config[uart].dev;
+    gpio_init_af(uart_config[uart].rx_pin, uart_config[uart].rxd, GPIO_IN);
+    gpio_init_af(uart_config[uart].tx_pin, uart_config[uart].txd, GPIO_OUT);
 
-#if UART_0_EN || UART_1_EN
     /* Enable clock for the UART while Running, in Sleep and Deep Sleep */
-    unsigned int uart_num = ( (uintptr_t)u - (uintptr_t)UART0 ) / 0x1000;
+    unsigned int uart_num = ( (uintptr_t)u - (uintptr_t)uart_config[uart].dev ) / 0x1000;
```

Comment:
`s/uart_config[uart].dev/u/`

Reply:
wait, ´u` is assigned with `uart_config[uart].dev` a few lines above, so this line seems to be evaluating always to `0`, right? This seems to be wrong.

Reply:
Ok, now I see: you have to leave this line unchanged, as it is computing the UART number from their base register offsets, and in this case `UART0` is needed here.